features/issues/api/use-get-issues.tsx

```diff
@@ -2,32 +2,36 @@ import { useEffect } from "react";
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { getIssues } from "@api/issues";
 import type { Page } from "@typings/page.types";
-import type { Issue } from "@api/issues.types";
+import type { Issue, IssuesParams } from "@api/issues.types";
 
 const QUERY_KEY = "issues";
 
-export function getQueryKey(page?: number) {
-  if (page === undefined) {
-    return [QUERY_KEY];
-  }
-  return [QUERY_KEY, page];
+export function getQueryKey(params: IssuesParams) {
+  const queryKey: (string | number)[] = [QUERY_KEY];
+  if (!params) return queryKey;
+  Object.values(params).map((param) => {
+    if (param) queryKey.push(param);
+  });
+  return queryKey;
```

Comment:
Just a nit: you could use a slightly different approach that might make the code more readable. Like

```
const paramKeys = Object.values(params).filter(Boolean);
return queryKey.conct(paramKeys);
```

I'm also not sure if this code might change the order of query keys. I think Object.values doesn't necessarily return an array in the same order for different objects. Not sure how to approach that. It would require some kind of reference. Maybe the keys of the params object could be sorted and then used for the iteration!? Anyway, not sure if that's even necessary